### PR TITLE
Add new parameter curl_timeout for curl_easy_setop

### PR DESCRIPTION
**Pull request by stweil**
Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

**Comment by stweil** [comment]
This addresses parts of feature request #3719.

File: src/api/baseapi.cpp [hunk 1]
```diff
@@ -99,6 +99,9 @@ namespace tesseract {
 
 static BOOL_VAR(stream_filelist, false, "Stream a filelist from stdin");
 static STRING_VAR(document_title, "", "Title of output document (used for hOCR and PDF output)");
+#ifdef HAVE_LIBCURL
+static INT_VAR(curl_timeout, 0, "Timeout for curl in seconds");
```
Comment (stweil) [inline src/api/baseapi.cpp:103]: Are there better suggestions for the name of the new parameter?

**Review by egorpugin — APPROVED** [review]

**Comment by egorpugin** [comment]
I think, we can refactor this code further. In the separate commit?
```
auto setopt = [error, curl](auto && ... args) {
        auto curlcode = curl_easy_setopt(curl, args...);
        if (curlcode != CURLE_OK) {
          return error("curl_easy_setopt");
        }
        return true;
};
if (!(1
&& setopt(CURLOPT_NOSIGNAL, 1L)
&& setopt(opt2, val2)
&& setopt(opt3, val3)
...
)) return false;
```

**Comment by stweil** [comment]
> I think, we can refactor this code further.

Yes, indeed. That can be done any time later.

**Pull request merged**